crates/ruma-events/src/room/server_acl.rs

```diff
@@ -45,14 +46,41 @@ impl RoomServerAclEventContent {
     pub fn new(allow_ip_literals: bool, allow: Vec<String>, deny: Vec<String>) -> Self {
         Self { allow_ip_literals, allow, deny }
     }
+
+    /// Returns true if and only if the server is allowed by the ACL rules.
+    pub fn is_allowed(&self, server_name: &ServerName) -> bool {
+        if !self.allow_ip_literals && server_name.is_ip_literal() {
+            return false;
+        }
+
+        for deny in &self.deny {
+            if let Ok(d) = glob::Pattern::new(&deny) {
```

Comment:
If the glob pattern fails, shouldn't we try to match the string directly anyways as well?

Reply:
I think the only time it could fail, but a literal comparison would still be useful is if the rule is an ipv6 address. I can implement that.